test/e2e/main_test.go

```diff
@@ -0,0 +1,66 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/client"
+)
+
+var (
+	kubeClient *kubeset.Clientset
+)
+
+func TestMain(m *testing.M) {
+	// if os.Getenv("KUBERNETES_CONFIG") == "" {
+	// 	kubeConfig := os.Getenv("KUBECONFIG")
+	// 	if kubeConfig == "" {
+	// 		kubeConfig = os.Getenv("HOME") + "/.kube/config"
+	// 	}
+	// 	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
+	// }
+
+	kubeconfig, err := client.GetConfig()
+	if err != nil {
+		fmt.Printf("unable to get kubeconfig: %s", err)
+		os.Exit(1)
+	}
+
+	kubeClient, err = kubeset.NewForConfig(kubeconfig)
+	if err != nil {
+		fmt.Printf("%#v", err)
+		os.Exit(1)
+	}
+
+	// e2e test job does not guarantee our operator is up before
+	// launching the test, so we need to do so.
+	fmt.Println("checking for console-operator availability")
+	err = waitForOperator()
+	if err != nil {
+		fmt.Println("failed waiting for operator to start")
+		os.Exit(1)
+	}
+
+	os.Exit(m.Run())
+}
+
+func waitForOperator() error {
+	depClient := kubeClient.AppsV1().Deployments(consoleapi.OpenShiftConsoleOperatorNamespace)
```

Comment:
Do we need to worry about the state of the deployment, perhaps for flakes?

Reply:
I think it's good practice to check if the resource, which functionality we are testing is checked for availability :)